### PR TITLE
Fix usage of `AliasGenerator` with `computed_field` decorator

### PR DESCRIPTION
**Pull request by sydney-runkle**
Fix https://github.com/pydantic/pydantic/issues/8768

Ensure that `AliasGenerator` is handled properly when using it along with a `computed_field` in a model.

Selected Reviewer: @samuelcolvin

### CODE REVIEW

File: pydantic/_internal/_generate_schema.py [hunk 1]
```diff
@@ -1009,6 +1009,49 @@ def _apply_alias_generator_to_field_info(
...
+        Args:
+            alias_generator: A callable that takes a string and returns a string, or an AliasGenerator instance.
+            computed_field_info: The ComputedFieldInfo instance to which the alias_generator is (maybe) applied.
+            computed_field_name: The name of the computed field from which to generate the alias.
+        """
+        # Apply an alias_generator if
+        # 1. An alias is not specified
+        # 2. An alias is specified, but the priority is <= 1
+
+        if alias_generator and (
```
Comment (dmontagu) [inline pydantic/_internal/_generate_schema.py:1029]: is this `if alias_generator` even necessary? It seems like it shouldn't be

File: tests/test_aliases.py [hunk 2]
```diff
@@ -702,3 +702,26 @@ class Foo(BaseModel):
...
+def test_alias_generator_with_computed_field_for_serialization() -> None:
+    """Tests that the alias generator is used for computed fields, with serialization_alias taking precedence over alias."""
+
+    class Rectangle(BaseModel):
+        model_config = ConfigDict(
+            alias_generator=AliasGenerator(
+                validation_alias=lambda field_name: f'{field_name}_val_alias',
+                alias=lambda field_name: f'{field_name}_alias',
+                serialization_alias=lambda field_name: f'{field_name}_ser_alias',
+            )
```
Comment (dmontagu) [inline tests/test_aliases.py:716]: Could we add a similar test with `serialization_alias=None` or whatever? And make sure it does the right thing using just the `alias`

Comment (sydney-runkle) [reply tests/test_aliases.py:716]: That test already exists:

```py
@pytest.mark.parametrize('alias_generator', upper_alias_generator)
def test_alias_generator_with_computed_field(alias_generator) -> None:
    class Rectangle(BaseModel):
        model_config = ConfigDict(populate_by_name=True, alias_generator=alias_generator)

        width: int
        height: int

        @computed_field
        @property
        def area(self) -> int:
            return self.width * self.height

    r = Rectangle(width=10, height=20)
    assert r.model_dump(by_alias=True) == {'WIDTH': 10, 'HEIGHT': 20, 'AREA': 200}
```

File: pydantic/_internal/_generate_schema.py [hunk 3]
```diff
@@ -1009,6 +1009,49 @@ def _apply_alias_generator_to_field_info(
...
+            # which supports the case where the alias_generator from a child class is used
+            # to generate an alias for a field in a parent class
+            if computed_field_info.alias_priority is None or computed_field_info.alias_priority <= 1:
+                computed_field_info.alias_priority = 1
+
+            # if the priority is 1, then we set the aliases to the generated alias
+            # note that we use the serialization_alias with priority over alias, as computed_field
+            # aliases are used for serialization only (not validation)
+            if computed_field_info.alias_priority == 1:
+                computed_field_info.alias = serialization_alias or alias
```
Comment (dmontagu) [inline pydantic/_internal/_generate_schema.py:1053]: I would do 
```suggestion
                computed_field_info.alias = serialization_alias if serialization_alias is not None else alias
```
to handle the case of `serialization_alias = ''` and `alias != ''`. Probably worth a comment about why, though not a test.

I could be convinced not to do that but I do think there are tests for the `alias = ''` case elsewhere in the codebase.

**Review by dmontagu — APPROVED** [review]

File: pydantic/_internal/_generate_schema.py (hunk 3 above)
Comment (sydney-runkle) [reply pydantic/_internal/_generate_schema.py:1053]: I think we should do this, but in a different PR, to both the field_info logic and the computed_field_info logic.

Comment (sydney-runkle) [reply pydantic/_internal/_generate_schema.py:1053]: Also, doesn't seem like an urgent change 👍
